src/rt/sched/behaviourcore.h

```diff
@@ -956,26 +995,29 @@ namespace verona::rt
         acquire_with_transfer(cown, transfer_count, ref_count);
 
         // Process writes without predecessor
-        if ((chain_had_no_predecessor) && (!curr_slot->is_read_only()))
+        if ((chain_had_no_predecessor) && (!chain_first_slot->is_read_only()))
         {
           if (cown->read_ref_count.try_write())
           {
             Logging::cout() << " Writer at head of queue and got the cown "
-                            << *curr_slot << Logging::endl;
+                            << *chain_first_slot << Logging::endl;
             ex_count++;
             yield();
           }
```

Comment:
Can `ex_count` be anything other than 0 in this branch?

```suggestion
          {
            Logging::cout() << " Writer at head of queue and got the cown "
                            << *chain_first_slot << Logging::endl;
            ec[first_body_index] += 1;
            yield();
            continue;
          }
```